octoprint_mrbeam/static/js/calibration/corner_calibration.js

```diff
@@ -426,7 +426,7 @@ $(function () {
         // USER CLICKS
         self.userClick = function (vm, ev) {
             // check if picture is loaded
-            if (window.location.href.indexOf("localhost") === -1)
+            if (window.location.getAttribute("href").indexOf("localhost") === -1)
```

Comment:
This seems quite fishy... Also why changing href for getAttribute? I think both would break in this case 🤔 
If we are just checking if we are running on the local server, we could do this instead:
`if (location.hostname === "localhost" || location.hostname === "127.0.0.1")`

Reply:
ah actually we are checking for the contrary 🤔 
So we are looking for localhost and if the index is -1 (Not found) then we enter the if...
We should extract this logic to a method but it would be nice to make it more readable like:
`if (location.hostname !== "localhost" || location.hostname !== "127.0.0.1")`
`
function isLocalhost(hostname) {
  hostname = location.hostname
  return hostname === "localhost" || isHomeIp(hostname)
}

function isHomeIP(ip) {
   let ipNumbers = ip.split('.');
   let isIpNumberLength = ipNumbers.length === 4;
   let isArrayOfNumbers = ipNumbers.every(element => {
     return !isNaN(element);
    });
   return isIpNumberLength && isArrayOfNumbers && ipNumbers[0] === '127';
}
`

Reply:
good catch, but I tried with the hostname and it was not working. I ned to revert this cahnge.
But I will try to understand why this is even needed